newrelic/resource_newrelic_cloud_azure_integrations.go

```diff
@@ -0,0 +1,2399 @@
+			},
+			"mysql": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The Azure mysql",
+				Elem:        cloudAzureIntegrationMysqlElem(),
+				MaxItems:    1,
+			},
+			"postgresql": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The Azure postgresql",
+				Elem:        cloudAzureIntegrationPostgresqlElem(),
+				MaxItems:    1,
+			},
+			"power_bi_dedicated": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The Azure powerBI dedicated",
+				Elem:        cloudAzureIntegrationPowerBiDedicatedElem(),
+				MaxItems:    1,
+			},
+			"redis_cache": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The Azure redis cache",
+				Elem:        cloudAzureIntegrationRedisCacheElem(),
+				MaxItems:    1,
+			},
+			"service_bus": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The Azure service bus",
+				Elem:        cloudAzureIntegrationServiceBusElem(),
+				MaxItems:    1,
+			},
+			"service_fabric": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The azure services fabric",
+				Elem:        cloudAzureIntegrationServiceFabricElem(),
+				MaxItems:    1,
+			},
+			"sql": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The Azure sql",
+				Elem:        cloudAzureIntegrationSQLElem(),
+				MaxItems:    1,
+			},
+			"sql_managed": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The Azure sql managed",
+				Elem:        cloudAzureIntegrationSQLManagedElem(),
+				MaxItems:    1,
+			},
+			"storage": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The Azure storage",
+				Elem:        cloudAzureIntegrationStorageElem(),
+				MaxItems:    1,
+			},
+			"virtual_machine": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The Azure virtual machine",
+				Elem:        cloudAzureIntegrationVirtualMachineElem(),
+				MaxItems:    1,
+			},
+			"virtual_networks": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The Azure virtual networks",
+				Elem:        cloudAzureIntegrationVirtualNetworksElem(),
+				MaxItems:    1,
+			},
+			"vms": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The Azure Vms",
+				Elem:        cloudAzureIntegrationVmsElem(),
+				MaxItems:    1,
+			},
+			"vpn_gateway": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The Azure vpn gateway",
+				Elem:        cloudAzureIntegrationVPNGatewayElem(),
+				MaxItems:    1,
+			},
+		},
+	}
+}
+
+func cloudAzureIntegrationSchemaBase() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"metrics_polling_interval": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "The data polling interval in seconds",
+		},
+		"resource_groups": {
```

Comment:
`resource_groups` is a list of strings. We'll want to change this schema and the way we expand and flatten it. Take a look at `aws_regions` in the AWS Integrations resource for an example.